ossindex/ossindex.go

```diff
@@ -106,68 +108,88 @@ func AuditPackages(purls []string) ([]types.Coordinate, error) {
 		return nil, err
 	}
 
-	if len(newPurls) > 0 {
-		var request types.AuditRequest
-		request.Coordinates = newPurls
-		var jsonStr, _ = json.Marshal(request)
+	var chunks = chunk(newPurls, MAX_COORDS)
 
-		req, err := setupRequest(jsonStr)
-		if err != nil {
-			return nil, err
-		}
+	for _, chunk := range chunks {
+		if len(chunk) > 0 {
+			var request types.AuditRequest
+			request.Coordinates = chunk
+			var jsonStr, _ = json.Marshal(request)
 
-		client := &http.Client{}
-		resp, err := client.Do(req)
-		if err != nil {
-			return nil, err
-		}
+			req, err := setupRequest(jsonStr)
+			if err != nil {
+				return nil, err
+			}
 
-		switch {
-		case strings.Contains(resp.Status, "200"):
-			fmt.Println(resp)
-		default:
-			return nil, errors.New("[" + resp.Status + "] error accessing OSS Index")
-		}
+			client := &http.Client{}
+			resp, err := client.Do(req)
+			if err != nil {
+				return nil, err
+			}
 
-		defer func() {
-			if err := resp.Body.Close(); err != nil {
-				log.Printf("error closing response body: %s\n", err)
+			switch {
+			case strings.Contains(resp.Status, "200"):
+				fmt.Println(resp)
+			default:
+				return nil, errors.New("[" + resp.Status + "] error accessing OSS Index")
 			}
-		}()
 
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return nil, err
-		}
+			defer func() {
+				if err := resp.Body.Close(); err != nil {
+					log.Printf("error closing response body: %s\n", err)
+				}
+			}()
 
-		// Process results
-		var coordinates []types.Coordinate
-		if err = json.Unmarshal([]byte(body), &coordinates); err != nil {
-			return nil, err
-		}
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
 
-		// Cache the new results
-		if err := db.Update(func(txn *badger.Txn) error {
-			for i := 0; i < len(coordinates); i++ {
-				var coord = coordinates[i].Coordinates
-				results = append(results, coordinates[i])
+			// Process results
+			var coordinates []types.Coordinate
+			if err = json.Unmarshal([]byte(body), &coordinates); err != nil {
+				return nil, err
+			}
 
-				var coordJson, _ = json.Marshal(coordinates[i])
+			// Cache the new results
+			if err := db.Update(func(txn *badger.Txn) error {
+				for i := 0; i < len(coordinates); i++ {
+					var coord = coordinates[i].Coordinates
+					results = append(results, coordinates[i])
 
-				err := txn.SetWithTTL([]byte(strings.ToLower(coord)), []byte(coordJson), time.Hour*12)
-				if err != nil {
-					return err
+					var coordJson, _ = json.Marshal(coordinates[i])
+
+					err := txn.SetWithTTL([]byte(strings.ToLower(coord)), []byte(coordJson), time.Hour*12)
+					if err != nil {
+						return err
+					}
 				}
-			}
 
-			return nil
-		}); err != nil {
-			return nil, err
+				return nil
+			}); err != nil {
+				return nil, err
+			}
 		}
 	}
 	return results, nil
 }
 
+func chunk(purls []string, chunkSize int) ([][]string) {
```

Comment:
You could do: `func chunk(purls []string, chunkSize int) (divided [][]string) {`

This will allow you to drop the `var divided [][]string` and also let you do a naked return at the end so just `return`.